drivers/storage/portworx/component/autopilot.go

```diff
@@ -131,6 +182,12 @@ func (c *autopilot) Reconcile(cluster *corev1.StorageCluster) error {
 	if err := c.createDeployment(cluster, ownerRef); err != nil {
 		return err
 	}
+
+	if IsOCPUserWorkloadSupported(c.k8sClient, c) {
+		if err := c.createSecret(cluster.Namespace, ownerRef); err != nil {
```

Comment:
Shouldn't we create the secret before we create the deployment? That way the chances of the AP pod failing are lower?